container/docker/docker.go

```diff
@@ -0,0 +1,144 @@
+package docker
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/client"
+	"github.com/wtsi-ssg/wr/container"
+	"github.com/wtsi-ssg/wr/math/conversion"
+)
+
+type Interactor struct {
+	dockerClient *client.Client
+}
+
+func NewInteractor(cl *client.Client) *Interactor {
+	return &Interactor{
+		dockerClient: cl,
+	}
+}
+
+// ContainerList is a docker specific implementation of the container package function
```

Comment:
More accurately, "ContainerList implements the Interactor interface method, which returns the list of containers."
Or don't say what it does. Or don't say what it implements.

Reply:
Done